### ISSUE TITLE
Failure to build vegetables

### ISSUE THREAD

**Issue by everythingfunctional** [fortran-lang/fpm#264]
I wanted to see how well the Fortran version was doing (I haven't really used it at all yet). I tried building my [vegetables](https://gitlab.com/everythingfunctional/vegetables) package and got the following error.

```
Unable to find source for module dependency: "parff" used by "app/make_driver_m.f90"
ERROR STOP 1

Error termination. Backtrace:
#0  0x105acfebd
#1  0x105ad0b75
#2  0x105ad1f6e
#3  0x105a03440
#4  0x105a9b886
```

Looks like it just isn't fetching the dependency specified for the executable, as seen in the below `fpm.toml` file.

```toml
name = "vegetables"
version = "6.0.0"
license = "MIT"
author = "Brad Richardson"
maintainer = "everythingfunctional@protonmail.com"
copyright = "2020 Brad Richardson"

[dependencies]
  iso_varying_string = { git = "https://gitlab.com/everythingfunctional/iso_varying_string.git", tag = "v2.0.0" }
  strff = { git = "https://gitlab.com/everythingfunctional/strff.git", tag = "v2.0.0" }

[[executable]]
  name = "make_vegetable_driver"
  source-dir = "app"
  main = "main.f90"
  [executable.dependencies]
    parff = { git = "https://gitlab.com/everythingfunctional/parff.git", tag = "v2.0.0" }
```

**Comment by awvwgk** [comment]
Duplicate of https://github.com/fortran-lang/fpm/issues/228

**Comment by awvwgk** [comment]
This issue might be fixed with #266 and #267 together.

**Comment by LKedward** [comment]
With #266 and #267 merged, there's a new error due to a bug in the source parsing routines which is incorrectly detecting a module as a program. I will look into a fix
